### PR TITLE
fix(library filters): disable library filters when there are no items in library

### LINKED ISSUE #724: Filter buttons can be used when libraries are empty
Regression introduced at #712

### PR DESCRIPTION
**Pull request by Camc314**
fixes #724

### CODE REVIEW

File: pages/library/_viewId.vue [hunk 1]
```diff
@@ -29,7 +29,7 @@
       <filter-button
         v-if="isSortable"
         :collection-info="collectionInfo"
-        :disabled="loading"
+        :disabled="loading || !(items.length && !hasFilters)"
```
Comment (ThibaultNocchi) [inline pages/library/_viewId.vue:32]: The logic here is wrong. If there are no items, it'll always be disabled. So you won't be able to clear the filters. I think it should be:
```suggestion
        :disabled="loading || (!items.length && !hasFilters)"
```

**Review by ThibaultNocchi — CHANGES_REQUESTED** [review]

File: pages/library/_viewId.vue (hunk 1 above)
Comment (Camc314) [reply pages/library/_viewId.vue]: Ah good catch! I also noticed that hasFilters was using `every` instead of `some`

**Review by ThibaultNocchi — APPROVED** [review]

File: locales/en-US.json [hunk 2]
```diff
@@ -109,6 +109,7 @@
   "latestLibrary": "Latest {libraryName}",
   "libraries": "Libraries",
   "libraryEmpty": "This library is empty",
+  "libraryEmptyFilters": "Remove some filters to show more items",
```
Comment (ferferga) [inline locales/en-US.json:112]:
```suggestion
  "libraryEmptyFilters": "Nothing matches the current filters. Remove some filters to show more items",
```

**Review by ferferga — CHANGES_REQUESTED** [review]

File: pages/library/_viewId.vue [hunk 3]
```diff
@@ -50,7 +50,7 @@
       <skeleton-item-grid v-if="loading" :view-type="viewType" />
       <item-grid :loading="loading" :items="items">
         <h1 v-if="!hasFilters && isDefaultView" class="text-h5">
-          {{ $t('libraryEmpty') }}
+          {{ hasFilters ? $t('libraryEmpty') : $t('libraryEmptyFilters') }}
```
Comment (ThibaultNocchi) [inline pages/library/_viewId.vue:53]: Wouldn't that be the opposite?

File: locales/en-US.json [hunk 4]
```diff
@@ -109,6 +109,7 @@
   "latestLibrary": "Latest {libraryName}",
   "libraries": "Libraries",
   "libraryEmpty": "This library is empty",
+  "libraryEmptyFilters": "No matched. Remove some filters to show more items",
```
Comment (ThibaultNocchi) [inline locales/en-US.json:112]: The first sentence feels wrong in English, I'd say something like "Nothing matched"

Comment (ferferga) [reply locales/en-US.json:112]: I guess he wants to say "No matches", but "No matched" is indeed wrong :)

Comment (Shadowghost) [reply locales/en-US.json:112]: What about "No matching items found"?

Comment (Camc314) [reply locales/en-US.json]: > The first sentence feels wrong in English, I'd say something like "Nothing matched"

Yeah that's a typo

File: locales/en-US.json [hunk 5]
```diff
@@ -109,6 +109,7 @@
   "latestLibrary": "Latest {libraryName}",
   "libraries": "Libraries",
   "libraryEmpty": "This library is empty",
+  "libraryEmptyFilters": "No matches Remove some filters to show more items",
```
Comment (ThibaultNocchi) [inline locales/en-US.json:112]: There's a point missing between the sentences ;)

File: locales/en-US.json (hunk 4 above)
Comment (ferferga) [reply locales/en-US.json]: @shadowghost The problem is that we have ``item`` repeated again in the same sentence, that's why I proposed a message without it.

**Review by ThibaultNocchi — APPROVED** [review]

**Review by ferferga — APPROVED** [review]

**Pull request merged**